diff -ur awstats-7.4-DIST/wwwroot/cgi-bin/awstats.pl awstats-7.4/wwwroot/cgi-bin/awstats.pl
--- awstats-7.4-DIST/wwwroot/cgi-bin/awstats.pl	2015-07-14 12:52:31.000000000 +0000
+++ awstats-7.4/wwwroot/cgi-bin/awstats.pl	2015-12-14 16:38:09.440198375 +0000
@@ -9162,7 +9162,7 @@
 				$i++;
 				push @fieldlib, 'tz';
 				$PerlParsingFormat .=
-"([^$LogSeparatorWithoutStar]+T[^$LogSeparatorWithoutStar]+)([-+\.]\\d\\d[:\\.\\dZ]*)";
+"([^$LogSeparatorWithoutStar]+T[^$LogSeparatorWithoutStar]+)(Z|[-+\.]\\d\\d[:\\.\\dZ]*)?";
 			}
 
 			# Special for methodurl and methodurlnoprot
Only in awstats-7.4/wwwroot/cgi-bin: awstats.pl.orig
Only in awstats-7.4/wwwroot/cgi-bin: awstats.pl.rej
