--- lib/ssluse.c.orig	2012-09-23 19:31:19.261986341 -0400
+++ lib/ssluse.c	2012-09-23 19:31:43.969809944 -0400
@@ -1476,18 +1476,8 @@
     use_sni(TRUE);
     break;
   case CURL_SSLVERSION_SSLv2:
-#ifdef OPENSSL_NO_SSL2
     failf(data, "OpenSSL was built without SSLv2 support");
     return CURLE_NOT_BUILT_IN;
-#else
-#ifdef USE_TLS_SRP
-    if(data->set.ssl.authtype == CURL_TLSAUTH_SRP)
-      return CURLE_SSL_CONNECT_ERROR;
-#endif
-    req_method = SSLv2_client_method();
-    use_sni(FALSE);
-    break;
-#endif
   case CURL_SSLVERSION_SSLv3:
 #ifdef USE_TLS_SRP
     if(data->set.ssl.authtype == CURL_TLSAUTH_SRP)
