--- virtual.rb.DIST	2016-09-25 16:34:58.531161946 +0000
+++ virtual.rb	2016-09-25 16:49:23.307231082 +0000
@@ -169,6 +169,17 @@
       return false
     end
     return true if in_lxc
+
+    # Alternative detection method - kernel 4.6+ with cgroup namespaces support
+    pathTty = Pathname.new('/dev/lxc/tty1')
+    if pathTty.exist? then
+      if pathTty.chardev? then
+        if File.stat('/dev/lxc/tty1').rdev_major == 136 then
+          return true
+        end
+      end
+    end
+
     return false
   end
 
