diff -urN ganglia-3.1.7.orig/libmetrics/linux/metrics.c ganglia-3.1.7/libmetrics/linux/metrics.c
--- ganglia-3.1.7.orig/libmetrics/linux/metrics.c	2010-02-17 06:05:39.000000000 -0500
+++ ganglia-3.1.7/libmetrics/linux/metrics.c	2012-09-24 17:47:43.451829986 -0400
@@ -1227,8 +1227,8 @@
 float find_disk_space(double *total_size, double *total_free)
 {
    FILE *mounts;
-   char procline[256];
-   char mount[128], device[128], type[32], mode[128];
+   char procline[1024];
+   char mount[1024], device[1024], type[1024], mode[1024];
    /* We report in GB = 1e9 bytes. */
    double reported_units = 1e9;
    /* Track the most full disk partition, report with a percentage. */
