diff -ur postfix-2.9.2.orig/src/local/mailbox.c postfix-2.9.2/src/local/mailbox.c
--- postfix-2.9.2.orig/src/local/mailbox.c	2012-01-25 01:40:34.000000000 +0100
+++ postfix-2.9.2/src/local/mailbox.c	2012-07-12 04:37:29.518422834 +0200
@@ -226,7 +226,7 @@
 	     defer_append : bounce_append)
 	    (BOUNCE_FLAGS(state.request), BOUNCE_ATTR(state.msg_attr));
     } else {
-	dsb_simple(why, "2.0.0", "delivered to mailbox");
+	dsb_simple(why, "2.0.0", "delivered to mailbox at %s", mailbox);
 	deliver_status = sent(BOUNCE_FLAGS(state.request),
 			      SENT_ATTR(state.msg_attr));
 	if (var_biff) {
diff -ur postfix-2.9.2.orig/src/local/maildir.c postfix-2.9.2/src/local/maildir.c
--- postfix-2.9.2.orig/src/local/maildir.c	2012-01-25 01:41:08.000000000 +0100
+++ postfix-2.9.2/src/local/maildir.c	2012-07-12 04:34:10.588422834 +0200
@@ -242,7 +242,7 @@
 	     defer_append : bounce_append)
 	    (BOUNCE_FLAGS(state.request), BOUNCE_ATTR(state.msg_attr));
     } else {
-	dsb_simple(why, "2.0.0", "delivered to maildir");
+	dsb_simple(why, "2.0.0", "delivered to maildir as %s", newfile);
 	deliver_status = sent(BOUNCE_FLAGS(state.request),
 			      SENT_ATTR(state.msg_attr));
     }
diff -ur postfix-2.9.2.orig/src/virtual/mailbox.c postfix-2.9.2/src/virtual/mailbox.c
--- postfix-2.9.2.orig/src/virtual/mailbox.c	2011-12-24 03:13:32.000000000 +0100
+++ postfix-2.9.2/src/virtual/mailbox.c	2012-07-12 04:41:20.698422834 +0200
@@ -154,7 +154,7 @@
 	    (BOUNCE_FLAGS(state.request),
 	     BOUNCE_ATTR(state.msg_attr));
     } else {
-	dsb_simple(why, "2.0.0", "delivered to mailbox");
+	dsb_simple(why, "2.0.0", "delivered to mailbox at %s", usr_attr.mailbox);
 	deliver_status = sent(BOUNCE_FLAGS(state.request),
 			      SENT_ATTR(state.msg_attr));
     }
diff -ur postfix-2.9.2.orig/src/virtual/maildir.c postfix-2.9.2/src/virtual/maildir.c
--- postfix-2.9.2.orig/src/virtual/maildir.c	2012-01-25 01:41:08.000000000 +0100
+++ postfix-2.9.2/src/virtual/maildir.c	2012-07-12 04:37:49.318422834 +0200
@@ -239,7 +239,7 @@
 	    (BOUNCE_FLAGS(state.request),
 	     BOUNCE_ATTR(state.msg_attr));
     } else {
-	dsb_simple(why, "2.0.0", "delivered to maildir");
+	dsb_simple(why, "2.0.0", "delivered to maildir as %s", newfile);
 	deliver_status = sent(BOUNCE_FLAGS(state.request),
 			      SENT_ATTR(state.msg_attr));
     }
